ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java

```diff
@@ -144,6 +146,21 @@ private Result handleAndReportFailure(final Block invalidBlock, final String rea
     return new Result(reason);
   }
 
+  private Result handleAndReportFailure(
+      final Block invalidBlock, final String reason, final BlockProcessor.Result result) {
+    if (result.causedBy().isPresent()) {
+      LOG.error(
+          "{}. Block {}, caused by {}", reason, invalidBlock.toLogString(), result.causedBy());
+      // TODO: if it's an internal error, don't add it
```

Comment:
is it for a next PR ?

Reply:
Yes. Right now I don't know what different causes we might encounter, or which ones should result in treating the block as bad.